src/app/shared/products-filter/products-filter.pipe.ts

```diff
@@ -0,0 +1,11 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Product} from '../products/product.interface';
+
+@Pipe({
+    name: 'productsFilter',
+})
+export class ProductsFilterPipe implements PipeTransform {
+    transform(products: Product[] | null, searchName: string): Product[] | undefined {
+        return products?.filter(p => p.name.includes(searchName));
```

Comment:
`p` -> `product`. В реальных проектах не используют сокращенные нейминги - считается плохим тоном) И бытует мнение у большинства разработчиков, что умение давать говорящие полные нейминги - показатели опытного разработчика. По этому я рекомендую сразу стараться привыкать давать полные и говорящие нейминги)

А вообще, тут я бы предложил сделать так
```suggestion
        return products?.filter(({name}) => name.includes(searchName));
```

Reply:
Привет, тут деструктурирующее присваивание связано с читабельности кода да?

Reply:
Да, хорошая пркактика, которая позволяет избавляться от постоянных цепочек `obj.key` в сторону `key`

Распространена еще такая практика для объявления констант:
```ts
const {keyFirst, keySecond} = obj;
```

Тоже помогает с читабельностью кода)